client/components/App.js

```diff
@@ -1,151 +1,137 @@
     }
     setIsJoining(false);
-  };
-
-  const handleLeaveClick = async () => {
-    await daily.destroy();
-    setDaily(null);
-    setUrl("");
-  };
+  }, []);
 
-  const [copied, setCopied] = useState(false);
-  const handleCopyURL = () => {
-    if (copy(url)) {
-      setCopied(true);
-      setTimeout(() => setCopied(false), 3000);
+  useEffect(() => {
+    if (query.url && typeof query.url === "string") {
+      joinRoom(query.url);
     }
+  }, [query]);
+
+  const handleSubmit = async (ev) => {
+    ev.preventDefault();
+    const roomUrl = ev.target.elements?.url?.value;
+    joinRoom(roomUrl);
   };
 
+  useEffect(
+    function initAppEffect() {
+      if (!url) return;
+      const initFrame = async () => {
+        if (DailyIframe.getCallInstance()) {
+          await DailyIframe.getCallInstance().destroy();
+        }
+        const frame = DailyIframe.createFrame(wrapperRef.current, {
+          showLeaveButton: true,
+          showUserNameChangeUI: true,
+          url,
+          customIntegrations: {
+            assistant: {
+              label: "AI Assistant",
+              location: "sidebar",
+              src:
+                location.protocol +
+                "//" +
+                location.host +
+                "/assistant?room_url=" +
+                url,
+            },
+          },
+          customTrayButtons: {
+            [robotBtnId]: getOpenRobotButton(),
+            [disableCCId]: getDisableCCButton(),
+          },
+        });
+        setDaily(frame);
+        await frame.join();
+
+        frame.once("left-meeting", () => {
+          frame.destroy();
+          setDaily(null);
+          setUrl("");
+        });
+      };
+      initFrame();
+    },
+    [url],
+  );
+
   return (
     <DailyProvider callObject={daily}>
       <div className="App">
         <h1>Daily AI Meeting Assistant Demo</h1>
         {url ? (
-          <>
-            <div className="actions">
-              <button disabled={copied} onClick={handleCopyURL}>
-                {copied ? "✅ Copied" : "📋 Copy room URL"}
-              </button>
-              <button onClick={handleLeaveClick}>🚪 Leave</button>
-            </div>
-          </>
+          <div className="actions">
+            <CopyRoomURLButton url={url} />
+          </div>
         ) : (
           <>
-            <p>Join the call and the AI Assistant bot joins automatically.</p>
+            <p>
+              Join the call and the AI Assistant bot joins and starts
+              transcription automatically.
+            </p>
             <p>
               Once there's enough context, you can ask the AI for a summary or
               other information, based on the spoken words.
             </p>
             <div>
-              <button disabled={isJoining} onClick={handleJoinClick}>
-                Join a call
-              </button>
+              <form onSubmit={handleSubmit}>
+                <input
+                  readonly={isJoining}
```

Comment:
I get an invalid DOM property exception on this property (should this be `readOnly`?)